catalog-rest-service/src/main/java/org/openmetadata/catalog/airflow/AirflowRESTClient.java

```diff
@@ -234,10 +235,15 @@ public HttpResponse<String> killIngestion(IngestionPipeline ingestionPipeline) {
   public Map<String, String> getHostIp() {
     HttpResponse<String> response;
     try {
-      response = getRequestAuthenticatedForJsonContent("%s/%s/ip", serviceURL, API_ENDPOINT);
-      if (response.statusCode() == 200) {
-        return JsonUtils.readValue(response.body(), new TypeReference<>() {});
+      if (this.hostIp == null || this.hostIp.isEmpty()) {
+        response = getRequestAuthenticatedForJsonContent("%s/%s/ip", serviceURL, API_ENDPOINT);
+        if (response.statusCode() == 200) {
+          return JsonUtils.readValue(response.body(), new TypeReference<>() {});
+        }
+      } else {
+        return Map.of("ip", this.hostIp);
       }
```

Comment:
I think this should be part of the parent class if it is the expected behavior for all the `PipelineServiceClient` objects.